pkg/yamlpath/path.go

```diff
@@ -143,6 +145,24 @@ func newPath(l *lexer) (*Path, error) {
 			return nil, err
 		}
 		return filterThen(filterLexemes, subPath), nil
+	case lexemePropertyName:
+		subPath, err := newPath(l)
+		if err != nil {
+			return nil, err
+		}
+		childName := strings.TrimPrefix(lx.val, ".")
+		childName = strings.TrimSuffix(childName, propertyName)
+		return propertyNameChildThen(childName, subPath), nil
+	case lexemeBracketPropertyName:
+		subPath, err := newPath(l)
+		if err != nil {
+			return nil, err
+		}
+		childNames := strings.TrimSpace(lx.val)
+		childNames = strings.TrimSuffix(childNames, propertyName)
+		childNames = strings.TrimSuffix(strings.TrimPrefix(childNames, "["), "]")
+		childNames = strings.TrimSpace(childNames)
```

Comment:
This code looks like it wouldn't work if it really has to deal with child names plural.

Reply:
This uses the same logic as the regular bracketchild syntax, it is taken care of by the bracketchildname function i added a test to verify